From 82c173abcefff0d11ae59294ffdbe06bebebf3ac Mon Sep 17 00:00:00 2001
From: Max Tottenham <max@tottenham.io>
Date: Tue, 19 May 2020 08:58:11 +0000
Subject: [PATCH 1/2] [PATCH] musl compat: Include <sys/time.h> in pytime.h for
 struct timeval def

  * pytime.h uses struct timeval* in the definition of functions without
    importing the header that defines struct timeval.
    This leads to redefinition errors like the following when attempting
    to compile pytime.c which *does* include <sys/time.h>:

pytime.c:506:1: error: conflicting types for __PyTime_AsTimeval_
  506 | _PyTime_AsTimeval(_PyTime_t t, struct timeval *tv, _PyTime_round_t round)
      | ^~~~~~~~~~~~~~~~~
In file included from /var/tmp/portage/dev-python/pypy3-exe-7.3.0/work/pypy3.6-v7.3.0-src/rpython/../pypy/module/cpyext/include/Python.h:84,
                 from pytime.c:1:
/var/tmp/portage/dev-python/pypy3-exe-7.3.0/work/pypy3.6-v7.3.0-src/rpython/../pypy/module/cpyext/include/pytime.h:121:17: note: previous declaration of __PyTime_AsTimeval_ was here
  121 | PyAPI_FUNC(int) _PyTime_AsTimeval(_PyTime_t t,
      |                 ^~~~~~~~~~~~~~~~~
---
 pypy/module/cpyext/include/pytime.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pypy/module/cpyext/include/pytime.h b/pypy/module/cpyext/include/pytime.h
index 158c460..40a16b8 100644
--- a/pypy/module/cpyext/include/pytime.h
+++ b/pypy/module/cpyext/include/pytime.h
@@ -2,6 +2,10 @@
 #ifndef Py_PYTIME_H
 #define Py_PYTIME_H
 
+#ifndef MS_WINDOWS
+#include <sys/time.h>
+#endif
+
 #include "pyconfig.h" /* include for defines */
 #include "object.h"
 
-- 
2.26.2

